cocotb/share/lib/gpi/GpiCommon.cpp

```diff
@@ -160,18 +161,34 @@ static void gpi_load_libs(std::vector<std::string> to_load)
          iter++)
     {
         void *lib_handle = NULL;
-        std::string full_name = "lib" + *iter + DOT_LIB_EXT;
+
+        std::string arg = *iter;
+
+        std::string lib_name;
+        std::string func_name;
+        auto it = std::find(arg.begin(), arg.end(), ':');
+        if (it == arg.end()) {
+            // no colon in the string, default
+            lib_name = arg;
+            func_name = arg + "_entry_point";
+        }
+        else {
+            lib_name = std::string(arg.begin(), it);
+            func_name = std::string(it+1, arg.end());
+        }
+
+        std::string full_name = "lib" + lib_name + DOT_LIB_EXT;
```

Comment:
```suggestion
        std::string full_name = lib_name + DOT_LIB_EXT;
```
Shouldn't we allow any library name here, not just `libXXX`?

Reply:
We can change but I would expect we want to keep default convention. Same way as you link with GCC you assume `lib` to be added.

Reply:
Agreed with themperek, I don't see much value to changing this.